### PR TITLE
Adding BLE support on Android Device + DPAD controller Connection

### PR DESCRIPTION
**Pull request by sparsh3dwe**

### CODE REVIEW

**Comment by thias15** [comment]
`> Task :app:testDebugUnitTest FAILED`

**Comment by sparsh3dwe** [comment]
> `> Task :app:testDebugUnitTest FAILED`

added null checks to fix the failures

**Comment by thias15** [comment]
Added a few more fixes. Also noticed that the jpeg images for the README are not pushed to this PR. Please add them after [compressing them](https://tinyjpg.com/).

**Review by thias15 — COMMENTED** [review]
README needs images and potentially some adjustments. Still need to test functionality - requires firmware from #329

File: android/app/src/main/java/org/openbot/autopilot/AutopilotFragment.java [hunk 1]
```diff
@@ -275,6 +275,8 @@ private void recreateNetwork(Model model, Network.Device device, int numThreads)
 
   @Override
   public synchronized void onResume() {
+    croppedBitmap = null;
+    tracker = null;
```
Comment (thias15) [inline android/app/src/main/java/org/openbot/autopilot/AutopilotFragment.java:279]: Why are these set to null here?

File: android/app/src/main/java/org/openbot/vehicle/Vehicle.java [hunk 2]
```diff
@@ -392,9 +392,11 @@ public boolean isUsbConnected() {
   private void sendStringToDevice(String message) {
     if (getConnectionType().equals("USB") && usbConnection != null) {
       usbConnection.send(message);
+      System.out.println("sanjeev usb = " + message);
```
Comment (thias15) [inline android/app/src/main/java/org/openbot/vehicle/Vehicle.java]: Please don't push debug messages.

File: android/app/src/main/java/org/openbot/autopilot/AutopilotFragment.java (hunk 1 above)
Comment (sparsh3dwe) [reply android/app/src/main/java/org/openbot/autopilot/AutopilotFragment.java:279]: So, there was a bug into the application, which is being fixed here by reinitializing the variables.

Bug was if you turn on the auto mode into the autopilot, then move to settings by clicking on the usb/ble icon and come back to the fragment. 

it crashes.

File: android/app/src/main/java/org/openbot/vehicle/Vehicle.java (hunk 2 above)
Comment (sparsh3dwe) [reply android/app/src/main/java/org/openbot/vehicle/Vehicle.java]: okay, will keep in mind

File: android/app/src/main/java/org/openbot/autopilot/AutopilotFragment.java (hunk 1 above)
Comment (thias15) [reply android/app/src/main/java/org/openbot/autopilot/AutopilotFragment.java:279]: Great 👍. One 🐛 less!

**Review by thias15 — COMMENTED** [review]
Looks good, tested. Just need to adjust README with updated text and screenshots.

**Pull request merged**